src/ServiceManagement/RecoveryServices/Commands.RecoveryServices/Service/InvokeAzureRecoveryServicesVaultUpgrade.cs

```diff
@@ -0,0 +1,238 @@
+                this.ValidateVaultUpgradePrerequisites();
+
+            bool actionConfirmed = false;
+            ResourceUpgradeDetails response = null;
+            string message = result == CheckVaultUpgradePrerequisitesResponse.Succeeded ?
+                Properties.Resources.PrerequisitesCheckPassed :
+                Properties.Resources.PrerequisitesCheckReturnedWarnings;
+
+            try
+            {
+                this.ConfirmAction(
+                    this.Force.IsPresent,
+                    string.Format(message),
+                    Properties.Resources.StartVaultUpgradeWhatIfMessage,
+                    this.VaultName,
+                    () =>
+                        {
+                            response = this.RecoveryServicesClient.StartVaultUpgrade(
+                                this.VaultName,
+                                this.Location,
+                                this.ResourceType,
+                                this.TargetResourceGroupName,
+                                this.Profile.Context.Subscription.Id.ToString());
+
+                            actionConfirmed = true;
+                        });
+
+                if (actionConfirmed)
+                {
+                    this.WriteObject(new ASRVaultUpgradeDetails(response));
+                    this.WaitForJobCompletion();
+                }
+            }
+            catch (Exception exception)
+            {
+                this.RecoveryServicesClient.HandleVaultUpgradeException(exception);
+            }
+        }
+
+        /// <summary>
+        /// Track resource upgrade operation.
+        /// </summary>
+        private void WaitForJobCompletion()
+        {
+            TrackResourceUpgradeResponse response = null;
+            DateTime startTime = DateTime.Now;
+            double taskTimeoutInSeconds = 6000;
+            double elapsedSeconds = 0;
+            ProgressRecord record =
+                new ProgressRecord(
+                    0,
+                    Properties.Resources.VaultUpgradeInProgress,
+                    Properties.Resources.WaitingForCompletion);
+
+            this.WriteResponse(Properties.Resources.VaultUpgradeInProgress);
+
+            do
+            {
+                Thread.Sleep(PSRecoveryServicesClient.TimeToSleepBeforeFetchingJobDetailsAgain);
+                response = this.RecoveryServicesClient.TrackVaultUpgrade(
+                    this.VaultName,
+                    this.Location,
+                    this.ResourceType);
+
+                elapsedSeconds = DateTime.Now.Subtract(startTime).TotalSeconds;
+                record.PercentComplete = (int)elapsedSeconds * 100 / (int)taskTimeoutInSeconds;
+                this.WriteProgress(record);
+            }
+            while (response.OperationStatus == Constants.InProgress &&
+                elapsedSeconds < taskTimeoutInSeconds &&
+                !this.StopProcessingFlag);
+
```

Comment:
This should make the progress bar disappear after upgrade, which is what you want.  If yoiu are running mocked, then this will mean no progress is displayedm, because you are not actually waiting any time between polls